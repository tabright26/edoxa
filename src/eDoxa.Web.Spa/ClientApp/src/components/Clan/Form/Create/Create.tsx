import React, { FunctionComponent } from "react";
import { FormGroup, Form } from "reactstrap";
import { Field, reduxForm } from "redux-form";
import Button from "components/Shared/Button";
import Input from "components/Shared/Input";
import { CREATE_CLAN_FORM } from "forms";
import { compose } from "recompose";
import FormValidation from "components/Shared/Form/Validation";

const validate = values => {
  var nameRegExp = new RegExp("^[a-zA-Z0-9- .,]{3,20}$");
  const errors: any = {};
  if (!values.name) {
    errors.name = "Name is required";
  } else if (!nameRegExp.test(values.name)) {
    errors.name =
      "Invalid format. Must between 3-20 characters and alphanumeric. Hyphens, spaces, dot and coma allowed.";
  }
  return errors;
};

const CustomForm: FunctionComponent<any> = ({
  handleSubmit,
  handleCancel,
  error
}) => (
  <Form onSubmit={handleSubmit}>
    {error && <FormValidation error={error} />}
    <Field
      type="text"
      name="name"
      label="Name"
      formGroup={FormGroup}
      component={Input.Text}
    />
    <FormGroup className="mb-0">
      <Button.Save className="mr-2" />
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

const enhance = compose<any, any>(
  reduxForm<any, { handleCancel: () => any }, string>({
    form: CREATE_CLAN_FORM,
    validate
  })
);

export default enhance(CustomForm);
