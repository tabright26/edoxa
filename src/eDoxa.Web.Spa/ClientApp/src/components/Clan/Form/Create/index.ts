import CustomForm from "./Create";

export default CustomForm;
