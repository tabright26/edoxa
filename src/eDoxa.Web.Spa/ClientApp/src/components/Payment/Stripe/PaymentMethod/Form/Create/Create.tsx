import React, { FunctionComponent } from "react";
import { Form } from "reactstrap";
import { reduxForm, InjectedFormProps } from "redux-form";
import {
  injectStripe,
  CardNumberElement,
  CardExpiryElement,
  CardCvcElement,
  ReactStripeElements
} from "react-stripe-elements";
import Button from "components/Shared/Button";
import { CREATE_STRIPE_PAYMENTMETHOD_FORM } from "utils/form/constants";
import { compose } from "recompose";
import { ValidationSummary } from "components/Shared/ValidationSummary";
import { attachStripePaymentMethod } from "store/actions/payment";
import {
  StripePaymentMethodsActions,
  ATTACH_STRIPE_PAYMENTMETHOD_FAIL
} from "store/actions/payment/types";
import { throwSubmissionError } from "utils/form/types";

interface FormData {}

type OutterProps = {
  handleCancel: () => void;
};

type InnerProps = ReactStripeElements.InjectedStripeProps &
  InjectedFormProps<FormData, Props>;

type Props = InnerProps & OutterProps;

const CreateStripePaymentMethodForm: FunctionComponent<Props> = ({
  handleSubmit,
  error,
  handleCancel
}) => (
  <Form onSubmit={handleSubmit}>
    <ValidationSummary error={error} />
    <dl className="row mb-0">
      <dt className="col-sm-4">Card number</dt>
      <dd className="col-sm-8">
        <CardNumberElement />
      </dd>
      <dt className="col-sm-4">Expiration date</dt>
      <dd className="col-sm-8">
        <CardExpiryElement />
      </dd>
      <dt className="col-sm-4">CVC</dt>
      <dd className="col-sm-8">
        <CardCvcElement />
      </dd>
      <dt className="col-sm-4 mb-0"></dt>
      <dd className="col-sm-8 mb-0">
        <Button.Save className="mr-2" />
        <Button.Cancel onClick={() => handleCancel()} />
      </dd>
    </dl>
  </Form>
);

const enhance = compose<InnerProps, OutterProps>(
  injectStripe,
  reduxForm<FormData, Props>({
    form: CREATE_STRIPE_PAYMENTMETHOD_FORM,
    onSubmit: (_values, dispatch: any, { stripe }) =>
      stripe.createPaymentMethod("card").then(result => {
        if (result.paymentMethod) {
          return dispatch(attachStripePaymentMethod(result.paymentMethod)).then(
            (action: StripePaymentMethodsActions) => {
              switch (action.type) {
                case ATTACH_STRIPE_PAYMENTMETHOD_FAIL: {
                  throwSubmissionError(action.error);
                  break;
                }
              }
            }
          );
        } else {
          return Promise.reject(result.error);
        }
      }),
    onSubmitSuccess: (_result, _dispatch, { handleCancel }) => handleCancel()
  })
);

export default enhance(CreateStripePaymentMethodForm);
