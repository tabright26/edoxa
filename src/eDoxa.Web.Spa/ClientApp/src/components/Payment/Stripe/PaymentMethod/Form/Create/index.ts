import CreateStripePaymentMethodContainer from "./Container";

export default CreateStripePaymentMethodContainer;
