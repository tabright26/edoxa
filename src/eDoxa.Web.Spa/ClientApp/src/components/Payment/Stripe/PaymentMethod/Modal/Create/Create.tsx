import React, { FunctionComponent } from "react";
import { Modal, ModalBody, ModalHeader } from "reactstrap";
import { connectModal } from "redux-modal";
import StripePaymentMethodForm from "components/Payment/Stripe/PaymentMethod/Form";
import { CREATE_STRIPE_PAYMENTMETHOD_MODAL } from "modals";
import { compose } from "recompose";

const CreateStripePaymentMethodModal: FunctionComponent<any> = ({ show, handleHide, type }) => (
  <Modal isOpen={show} toggle={handleHide}>
    <ModalHeader toggle={handleHide}>ADD NEW PAYMENT METHOD</ModalHeader>
    <ModalBody>
      <StripePaymentMethodForm.Create type={type} handleCancel={() => handleHide()} />
    </ModalBody>
  </Modal>
);

const enhance = compose<any, any>(connectModal({ name: CREATE_STRIPE_PAYMENTMETHOD_MODAL }));

export default enhance(CreateStripePaymentMethodModal);
