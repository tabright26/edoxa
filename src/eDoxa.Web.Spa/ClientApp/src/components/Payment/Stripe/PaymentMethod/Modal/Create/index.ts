import CustomModal from "./Create";

export default CustomModal;
