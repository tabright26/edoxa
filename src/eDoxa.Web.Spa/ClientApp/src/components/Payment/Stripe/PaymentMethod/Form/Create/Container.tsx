import { connect } from "react-redux";
import { injectStripe } from "react-stripe-elements";
import { attachStripePaymentMethod } from "store/actions/payment";
import {
  ATTACH_STRIPE_PAYMENTMETHOD_FAIL,
  StripePaymentMethodsActions
} from "store/actions/payment/types";
import Create from "./Create";
import { compose } from "recompose";
import { throwSubmissionError } from "utils/form/types";

const mapDispatchToProps = (dispatch: any, ownProps: any) => {
  return {
    createStripePaymentMethod: () =>
      ownProps.stripe.createPaymentMethod(ownProps.type).then(result => {
        if (result.paymentMethod) {
          return dispatch(
            attachStripePaymentMethod(result.paymentMethod.id)
          ).then((action: StripePaymentMethodsActions) => {
            switch (action.type) {
              case ATTACH_STRIPE_PAYMENTMETHOD_FAIL: {
                throwSubmissionError(action.error);
                break;
              }
            }
          });
        } else {
          return Promise.reject(result.error);
        }
      })
  };
};

const enhance = compose<any, any>(
  injectStripe,
  connect(null, mapDispatchToProps)
);

export default enhance(Create);
