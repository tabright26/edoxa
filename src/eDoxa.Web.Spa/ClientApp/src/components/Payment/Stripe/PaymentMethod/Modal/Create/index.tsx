import React, { FunctionComponent } from "react";
import { Modal, ModalBody, ModalHeader } from "reactstrap";
import { connectModal, InjectedProps } from "redux-modal";
import StripePaymentMethodForm from "components/Payment/Stripe/PaymentMethod/Form";
import { CREATE_STRIPE_PAYMENTMETHOD_MODAL } from "utils/modal/constants";
import { compose } from "recompose";
import { ReactStripeElements, Elements } from "react-stripe-elements";
import { ModalSubtitle } from "components/Shared/Modal/Subtitle";

type InnerProps = InjectedProps & ReactStripeElements.InjectedStripeProps;

type OutterProps = {};

type Props = InnerProps & OutterProps;

const Create: FunctionComponent<Props> = ({ show, handleHide }) => (
  <Modal backdrop="static" centered isOpen={show} toggle={handleHide}>
    <ModalHeader className="my-auto bg-gray-900" toggle={handleHide}>
      <span className="d-block text-uppercase">ADD NEW PAYMENT METHOD</span>
      <ModalSubtitle>We accept all credit card</ModalSubtitle>
    </ModalHeader>
    {show && (
      <ModalBody>
        <Elements>
          <StripePaymentMethodForm.Create handleCancel={() => handleHide()} />
        </Elements>
      </ModalBody>
    )}
  </Modal>
);

const enhance = compose<InnerProps, OutterProps>(
  connectModal({
    name: CREATE_STRIPE_PAYMENTMETHOD_MODAL,
    destroyOnHide: false
  })
);

export default enhance(Create);
