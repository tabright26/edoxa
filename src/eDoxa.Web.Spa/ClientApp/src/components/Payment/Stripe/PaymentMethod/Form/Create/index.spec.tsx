import React from "react";
import Create from ".";
import { configureStore } from "store";
import { Provider } from "react-redux";

const shallow = global["shallow"];

const store = configureStore();

//  const createWrapper = (): ReactWrapper | any => {
//    return mount(
//     <Provider store={store}>
//       <Create />
//     </Provider>;
//    );
// };

describe("<PaymentMethodCreateForm />", () => {
  it("should match the snapshot", () => {
    const shallowWrapper = shallow(
      <Provider store={store}>
        <Create />
      </Provider>
    );
    expect(shallowWrapper).toMatchSnapshot();
  });

  // describe("defines paymentMethod create form fields", () => {
  //   it("renders save button", () => {
  //     const wrapper = createWrapper();
  //     const saveButton = wrapper.find("SaveButton").first();
  //     const button = saveButton.find("button").first();

  //     expect(button.prop("type")).toBe("submit");
  //     expect(button.text()).toBe("Save");
  //   });

  //   it("renders cancel button", () => {
  //     const wrapper = createWrapper();
  //     const cancelButton = wrapper.find("CancelButton").first();
  //     const button = cancelButton.find("button").first();

  //     expect(button.prop("type")).toBe("button");
  //     expect(button.text()).toBe("Cancel");
  //   });
  // });
});
