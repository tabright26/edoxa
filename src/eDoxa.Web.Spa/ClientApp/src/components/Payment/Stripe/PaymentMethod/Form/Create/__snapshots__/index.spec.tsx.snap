// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentMethodCreateForm /> should match the snapshot 1`] = `ShallowWrapper {}`;
